src/nvim/highlight_group.c

```diff
@@ -232,6 +232,11 @@ static const char *highlight_init_both[] = {
   "default link DiagnosticVirtualTextInfo  DiagnosticInfo",
   "default link DiagnosticVirtualTextHint  DiagnosticHint",
   "default link DiagnosticVirtualTextOk    DiagnosticOk",
+  "default link DiagnosticVirtualLinesError DiagnosticError",
+  "default link DiagnosticVirtualLinesWarn  DiagnosticWarn",
+  "default link DiagnosticVirtualLinesInfo  DiagnosticInfo",
+  "default link DiagnosticVirtualLinesHint  DiagnosticHint",
+  "default link DiagnosticVirtualLinesOk    DiagnosticOk",
```

Comment:
If we really need new hl groups they should probably link to `DiagnosticVirtualText*` ? I don't really see why we need new hl groups though.

Reply:
Conceivably, you might wish to use, e.g., italics for virtual text (to more clearly distinguish it from actual buffer text), while this would be too much for virtual lines (where you potentially see more text, including all diagnostics).

Conversely, you might wish to use a different background to make clear these are boguslines.
(Although that would probably be better handled by allowing for a uniform background color different from normal.)

Reply:
I do think that we need separate highlight groups. @clason gave some good examples in favor of it.

> If we really need new hl groups they should probably link to `DiagnosticVirtualText*` ? I don't really see why we need new hl groups though.

Sure, but I find it a bit weird that this would be the only diagnostic handler whose highlight groups don't link to `Diagnostic*`. I chose this for a matter of consistency.